infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -113,25 +113,25 @@ public LogReplicationConfigManager(CorfuRuntime runtime, ServerContext serverCon
         this.pluginConfigFilePath = serverContext == null ? EMPTY_STR : serverContext.getPluginConfigFilePath();
         initLogReplicationVersionPlugin(runtime);
         setupVersionTable();
-        config = generateConfig();
+        subscriberToConfigMap = generateConfig();
     }
 
-    private LogReplicationConfig generateConfig() {
+    private Map<ReplicationSubscriber, LogReplicationConfig> generateConfig() {
         PersistentCorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> registryTable =
             runtime.getTableRegistry().getRegistryTable();
-        registryTableEntries = registryTable.entryStream().collect(Collectors.toList());
 
-        Map<ReplicationSubscriber, Set<String>> replicationSubscriberToStreamsMap = new HashMap<>();
+        List<Map.Entry<TableName, CorfuRecord<TableDescriptors, TableMetadata>>> registryTableEntries =
+            registryTable.entryStream().collect(Collectors.toList());
+
         Map<UUID, List<UUID>> streamToTagsMap = new HashMap<>();
+        Set<String> streamsToReplicate = new HashSet<>();
+        Set<UUID> streamsToDrop = registryTable.entryStream().map(entry -> getFullyQualifiedTableName(entry.getKey()))
+            .map(CorfuRuntime::getStreamID).collect(Collectors.toSet());
 
         registryTableEntries.forEach(entry -> {
 
             if (entry.getValue().getMetadata().getTableOptions().getIsFederated()) {
-                ReplicationSubscriber subscriber = ReplicationSubscriber.getDefaultReplicationSubscriber();
-                Set<String> streamsToReplicate =
-                    replicationSubscriberToStreamsMap.getOrDefault(subscriber, new HashSet<>());
```

Comment:
why is this mapping removed/replaced with a set? We will need the 2 maps(subscriber-> streamsToReplicate & subscriber -> streamToDrop), right?

Reply:
This is from @pankti-m 's change but my understanding from her PR is that since `LogReplicationConfigManager` has the mapping of subscriber->LogReplicationConfig and LogReplicationConfig has the streamsToReplicate, she is removing it from the Subscriber since the LogReplicationConfig now carries that mapping.